Dockerfile

```diff
@@ -16,20 +16,32 @@
 FROM golang:1.18 AS builder
 
 WORKDIR /build-dir
-# Copy the go source
-COPY . .
-# cache deps before building and copying source so that we don't need to re-download as much
+
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# Cache dependencies before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
 USER 0
 RUN go mod download
+
+# Copy the go source 
+COPY Makefile Makefile
+COPY bindata.mk bindata.mk
+COPY pkg/ pkg/
+COPY cmd/ cmd/
+COPY hack/ hack/
+COPY test/ test/
+
+# Build the kcp-tests binary
 RUN make build
 
 FROM quay.io/centos/centos:stream8
```

Comment:
@Phaow should we use more of a RHEL base Image instead of centos ?

Reply:
Hi @kasturinarra , we are in an public repo, from my understanding we'd better use a opensource image. RHEL seems for commercial usage. I didn't check more about RHEL docker image, do you have the name+tag?

Reply:
@Phaow how about this ? https://hub.docker.com/r/redhat/ubi8/tags

Reply:
Good suggestion. I think we could consider it. I'll do some test and update later maybe next week plan. Seems we could simply migrate centos os base image to ubi8, tbh, I'll also build the image and test a round.